Sources/SwiftFrameCore/Extensions/Collection+Extensions.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+
+extension Array where Element == URL {
+
+    func filter(pattern: String?) throws -> Self {
+        guard let pattern = pattern else {
+            return self
+        }
+        guard let regex = try? NSRegularExpression(pattern: pattern, options: .caseInsensitive) else {
+            throw NSError(description: "Invalid regular expression: \"\(pattern)\"")
+        }
+        return self.filter { url in
+            let lastComponent = url.deletingPathExtension().lastPathComponent
```

Comment:
Method name could maybe be concretised. `filter` makes it sound as if you're filtering on the whole `URL` when you're just filtering on the last path component without extension.